Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -770,19 +760,16 @@ private async void QueryResults()
                 }
             }, currentCancellationToken, TaskContinuationOptions.NotOnCanceled, TaskScheduler.Default);
 
-            // plugins is ICollection, meaning LINQ will get the Count and preallocate Array
-
-            var tasks = plugins.Select(plugin => plugin.Metadata.Disabled switch
-            {
-                false => QueryTask(plugin),
-                true => Task.CompletedTask
-            }).ToArray();
-
-
             try
             {
-                // Check the code, WhenAll will translate all type of IEnumerable or Collection to Array, so make an array at first
-                await Task.WhenAll(tasks);
+                await Parallel.ForEachAsync(plugins, currentCancellationToken,
+                    async (pair, token) =>
+                    {
+                        await Task.Delay(pair.Metadata.SearchDelay ?? Settings.SearchDelay, token);
+                        if (pair.Metadata.Disabled)
+                            return;
+                        await QueryTask(pair, token);
```

Comment:
Should we check `pair.Metadata.Disabled` before delay?

Reply:
I agree we should, Though I don't think it's a really big deal.